python/test/gluon/test_core.py

```diff
@@ -783,3 +783,31 @@ def tmem_copy_no_scales(in_ptr, out_ptr, M: ttgl.constexpr, N: ttgl.constexpr, B
 
     tmem_copy_no_scales[(1, )](input, output, M, N, BLOCK_N, swizzle, num_warps=num_warps)
     assert (output == input).all()
+
+
+@gluon.jit
+def early_return_kernel(x):
+    if x.sum(0).sum(0):
+        return x
+    x = x + x
+    print("value", x)
+    return x
+
+
+def test_2d_tensor_early_return():
+    warp_size = ttgl.constexpr(THREADS_PER_WARP)
+
+    @gluon.jit
+    def kernel(N):
+        layout: ttgl.constexpr = ttgl.BlockedLayout([1, 1], [1, warp_size], [1, 4], [1, 0])
+        BLOCK: ttgl.constexpr = 32
+
+        x0 = ttgl.arange(0, BLOCK, layout=ttgl.SliceLayout(1, layout))
+        x1 = ttgl.arange(0, BLOCK, layout=ttgl.SliceLayout(0, layout))
+        x = x0[:, None] * x1[None, :]
+        for i in range(N):
+            x += early_return_kernel(x)
+        print("value", x)
+
+    compiled_kernel = kernel.warmup(N=100, grid=(1, ))
+    assert compiled_kernel.asm["llir"].count("define") == 1
```

Comment:
The kernel uses `print` inside so we will see a bunch of print related `__ockl` internal function `define`s for AMD side. Maybe we can check the `early_return_kernel` symbol here, or avoid using `print` in the kernel.

Reply:
right, maybe just adding a store to make sure x is not dead is enough